runtime/compiler/ilgen/Walker.cpp

```diff
@@ -7219,9 +7229,9 @@ TR_J9ByteCodeIlGenerator::storeInstance(int32_t cpIndex)
    // code to handle volatiles moved to CodeGenPrep
    //
    TR::Node * node;
-   if (type == TR::Address && _generateWriteBarriers)
+   if ((type == TR::Address && _generateWriteBarriers) || comp()->getOption(TR_EnableFieldWatch))
```

Comment:
We seem to be orthogonal here w.r.t. write and read barriers. Write barriers are controlled via `_generateWriteBarriers` which is initialized once in the constructor while we check for read barriers via `TR_EnableFieldWatch` every time. Not only this but we seem to store the result of the `getOption` call in a `generateReadBarrier` variable above, but here we query it directly.

I'm not a fan of this inconsistency as it makes the code harder to read and follow. Can we stick to one approach? Are we able to introduce a `_generateReadBarriers` field analogous to `_generateWriteBarriers`?

Reply:
The other `_generateWriteBarriers` is for `comp->getOptions()->getGcMode() != TR_WrtbarNone` and wrtbar are only needed for address type in this case so one bool field for all wrtbars won't work.  I can change it to  `_generateWriteBarriersForGC` to be more clear and create another field called `_generateReadBarrierForFieldWatch` and `_generateWriteBarrierForFieldWatch` and all of them got initialized once at constructor.

Reply:
Sounds like a reasonable, and much more consistent approach.

Reply:
changed the existing field to be  `_generateWriteBarriersForGC` 
 and  added `_generateReadBarrierForFieldWatch` `_generateWriteBarrierForFieldWatch`